drivers/sensor/lis2mdl/lis2mdl.c

```diff
@@ -294,8 +360,11 @@ static int lis2mdl_init(const struct device *dev)
 {
 	struct lis2mdl_data *lis2mdl = dev->data;
 	uint8_t wai;
+	int rc = 0;
 
 	lis2mdl->dev = dev;
+	lis2mdl->single_mode = IS_ENABLED(DT_INST_PROP(0, single_mode));
```

Comment:
these settings should be stored directly in driver config (`dev->config`) so that they will have the right value on each instance.